Dockerfile

```diff
@@ -0,0 +1,63 @@
+# syntax=docker/dockerfile:1
+
+# Comments are provided throughout this file to help you get started.
+# If you need more help, visit the Dockerfile reference guide at
+# https://docs.docker.com/engine/reference/builder/
+
+################################################################################
+# Create a stage for building the application.
+ARG GO_VERSION=1.21
+FROM golang:${GO_VERSION} AS build
+WORKDIR /src
+
+COPY . /src
+
+# Download dependencies as a separate step to take advantage of Docker's caching.
+# Leverage a cache mount to /go/pkg/mod/ to speed up subsequent builds.
+# Leverage bind mounts to go.sum and go.mod to avoid having to copy them into
+# the container.
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+    --mount=type=bind,source=go.sum,target=go.sum \
+    --mount=type=bind,source=go.mod,target=go.mod \
+    go mod download -x
+
+# Build the application.
+# Leverage a cache mount to /go/pkg/mod/ to speed up subsequent builds.
+# Leverage a bind mount to the current directory to avoid having to copy the
+# source code into the container.
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+    --mount=type=bind,target=. \
+    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -o /generator
+
+RUN ./generator
```

Comment:
This way, you will generate the keys only once during the build, and then everyone who will use that image will have the same values of the keys.

You should rather only build binary at this stage and copy that binary in the next stage and run it as a part of the entrypoint. (so the best would be to have the script that at first will run the binary and then create a secret).

Reply:
Changed to execute when container is running. Thanks